### PR TITLE
Accept NTLM and Negotiate as valid authentication methods

### LINKED ISSUE #2974: NTLM Authentication broken in Alamofire 5
* [x] I've read, understood, and done my best to follow the [CONTRIBUTING guidelines](https://github.com/Alamofire/Alamofire/blob/master/CONTRIBUTING.md).

## What did you do?

Tried to perform an authenticated request to a server that requires NTLM authentication.

### What did you expect to happen?

I expected the request to succeed, as it did in Alamofire 4.

### What happened instead?

Request failed, Alamofire didn't even try to use the provided credentials. This is a regression.

### Alamofire Environment

**Alamofire version:** 5.0.0-rc.2
**Xcode version:** 11.1
**Swift version:** 5.1
**Platform(s) running Alamofire:** iOS
**macOS version running Xcode:** 10.15

## Detailed Explanation

I did some digging around and found that Alamofire no longer treats `NSURLAuthenticationMethodNTLM` treating as a valid authentication method: 

https://github.com/Alamofire/Alamofire/blob/6afd00ef8bf800edd14cc141e0677045b5d02a67/Source/SessionDelegate.swift#L78-L88

I'll open a PR adding the missing cases to that switch as an attempt to resolve this.

## Demo Project

Unfortunately I was unable to find any working NTLM auth server sandbox on the Internet. Let me know a demo project is crucial here, I'll figure a way to provide it without publicly disclosing my case.

### PR DESCRIPTION
**Pull request by akashivskyy**
### Issue Link

Fixes #2974.

### Goals

This PR fixes a regression introduced in Alamofire 5 where NTLM and Negotiate authentication methods were no longer accepted.

### Testing Details

Not sure how to write a test for this — I didn't find any sandbox server (incl. httpbin) that would allow to test NTLM auth, but I can confirm this resolves the issue for me.

### CODE REVIEW

**Comment by jshier** [comment]
Nice catch! I didn't realize we implicitly supported these before. PR looks good, but since we're handling more than HTTP authentication, the `attemptHTTPAuthentication` method should probably updated to something like `attemptCredentialAuthentication`, along with the doc comment on the method. Once that's updated, I think we're good to merge.

**Comment by jshier** [comment]
If you pull the changes from `master`, you'll get the GitHub Actions CI builds, which should be more reliable than Travis CI.

**Comment by akashivskyy** [comment]
@jshier Done!

File: Source/SessionDelegate.swift [hunk 1]
```diff
@@ -121,14 +121,14 @@ extension SessionDelegate: URLSessionTaskDelegate {
...
-    /// Evaluates the HTTP authentication `URLAuthenticationChallenge` received for `task`.
+    /// Evaluates the credential-based authentication `URLAuthenticationChallenge` received for `task`.
     ///
     /// - Parameters:
     ///   - challenge: The `URLAuthenticationChallenge`.
     ///   - task:      The `URLSessionTask` which received the challenge.
     ///
     /// - Returns:     The `ChallengeEvaluation`.
-    func attemptHTTPAuthentication(for challenge: URLAuthenticationChallenge,
+    func attemptCredentialAuthentication(for challenge: URLAuthenticationChallenge,
```
Comment (jshier) [inline Source/SessionDelegate.swift]: Please adjust the whitespace here. You can do it manually, or just run `swiftformat .` from the root of the project.

File: Source/SessionDelegate.swift [hunk 2]
```diff
@@ -121,14 +121,14 @@ extension SessionDelegate: URLSessionTaskDelegate {
...
                                    belongingTo task: URLSessionTask) -> ChallengeEvaluation {
```
Comment (jshier) [inline Source/SessionDelegate.swift]: Please adjust the whitespace here to align with the line above. You can do it manually or run `swiftformat .` at the root of the project.

**Comment by akashivskyy** [comment]
The test failure seems unrelated. 🤔

**Pull request merged**